crates/block-producer/src/mempool/dependency_graph.rs

```diff
@@ -61,6 +61,20 @@ pub struct DependencyGraph<K, V> {
     processed: BTreeSet<K>,
 }
 
+impl<K, V> Debug for DependencyGraph<K, V>
+where
+    K: Debug,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("DependencyGraph")
+            .field("pending", &self.pending)
+            .field("vertices", &self.vertices.keys())
+            .field("processed", &self.processed)
+            .field("roots", &self.roots)
+            .finish()
```

Comment:
This debug impl elides data and graph edges. Data would include transaction data which makes this debug impossible to navigate.

I'm unsure if we shouldn't add it all anyway.

Reply:
My two cents: I think I'd add it for completeness since it seems like potentially valuable debug information unless it interrupts an existing workflow somehow. Alternatives could be to have a dedicated method that hides that data or viceversa (and maybe if it's decided that it's important enough, a feature or env variable could help too).

Reply:
I've added the graph edges, but kept the data itself out under the assumption that debugging the graph should just show the graph structure and not the node contents.